docs/howto/operate/add-aws-hub.md

```diff
@@ -0,0 +1,222 @@
+
+```bash
+mv <cluster_name>.key ssh-keys/<cluster_name>.key
+mv <cluster_name>.key.pub ssh-keys/<cluster_name>.key.pub
+```
+
+and encrypt the private key with
+
+```bash
+sops -i -e ssh-keys/<cluster_name>.key
+```
+
+before pushing your changes to the repository.
+
+### Apply workaround to run CoreDNS on the master node
+
+```{seealso}
+More details [in this issue](https://github.com/kubernetes/kops/issues/11199)
+```
+
+1. Patch CoreDNS with the following two commands
+
+```bash
+kubectl -n kube-system patch deployment kube-dns --type json --patch '[{"op": "add", "path": "/spec/template/spec/tolerations", "value": [{"key": "node-role.kubernetes.io/master", "effect": "NoSchedule"}]}]'
+deployment.apps/kube-dns patched
+
+kubectl -n kube-system patch deployment kube-dns-autoscaler --type json --patch '[{"op": "add", "path": "/spec/template/spec/tolerations", "value": [{"key": "node-role.kubernetes.io/master", "effect": "NoSchedule"}]}]'
+```
+
+2. After applying those patches, you should be able to validate the cluster with
+
+```bash
+kops validate cluster --wait 10m
+```
+
+### Create an EFS for your cluster
+
+1. Install `boto3` with pip or conda
+
+2. Create an [EFS](https://aws.amazon.com/efs/) file system for this hub with
+
+```bash
+python3 setup-efs.py <cluster_name>hub.k8s.local <region-of-cluster>
+```
+
+This will output an fs-<xxxxxxxx> id. You should use that value 
+(it should be something like `fs-<id>.efs.<region>.amazonaws.com`) in
+the `basehub.nfsPVC.nfs.serverIP` at you hub config file. 
+
+## Deploy the new hub
+
+1. First, `cd` back to the root of the repository
+
+2. Generate a kubeconfig that will be used by the deployer with
+
+```bash
+KUBECONFIG=secrets/<cluster_name>.yaml kops export kubecfg --admin=730h <cluster_name>hub.k8s.local
+```
+
+3. Encrypt (in-place) the generated kubeconfig with sops
+
+```bash
+sops -i -e secrets/<cluster_name>.yaml
+```
+
+4. Generate a new config file for your cluster
+
+You can use of the existing cluster config files as a "template" for your cluster (for example, [here is the Farallon Institute config file](https://github.com/2i2c-org/pilot-hubs/blob/master/config/hubs/farallon.cluster.yaml)).
+You may need to tweak names, `serverIP` and singleuser's images references. Make sure 
+you set up the `profileList` section to be compatible with your kops cluster (ie. match 
+the `node_selector` with the proper `instance-type`).
+
+5. Set `proxy.https.enabled` to `false`. This creates the hubs without trying to give 
+them HTTPS, so we can appropriately create DNS entries for them.
+
+6. Deploy the hub (or hubs in case you are deploying more than one) without running the
+test with
+
+```bash
+python3 deployer deploy <cluster_name> --skip-hub-health-test
+```
+
+7. Get the AWS external IP for your hub with (supposing your hub is `staging`):
+
+```bash
+kubectl -n staging get svc proxy-public
+```
+
+Create a CNAME record for `staging.foo.2i2c.cloud` and point it to the AWS external IP.
```

Comment:
This needs to be repeated for staging and prod as well, no?

Reply:
Yes, I will make a note about that as well.